src/formatDate.js

```diff
@@ -9,6 +9,38 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
-}
+  const data = date.split(fromFormat[3]);
+
+  const object = {};
+
+  for (let i = 0; i < 3; i++) {
+    object[fromFormat[i]] = data[i];
+  }
+
+  if (object.YYYY) {
+    object.YY = object.YYYY.slice(2);
+  }
+
+  if (object.YY < 30) {
+    object.YYYY = '20' + `${object.YY}`;
+  } else {
+    object.YYYY = '19' + `${object.YY}`;
+  }
```

Comment:
The logic for converting from `YY` to `YYYY` is outside of any conditional that checks if the `YY` format is actually present in the `fromFormat`. This could lead to incorrect conversions if the `fromFormat` is already `YYYY` or if the year part is not included at all.